data-prepper-api/src/main/java/org/opensearch/dataprepper/model/configuration/PipelinesDataFlowModel.java

```diff
@@ -28,6 +30,11 @@ public class PipelinesDataFlowModel {
     @JsonInclude(JsonInclude.Include.NON_NULL)
     private DataPrepperVersion version;
 
+    @JsonProperty("pipeline_extensions")
```

Comment:
In #2590 this is proposed as `pipeline_configurations`. I tend to think that we should keep this original name.

There are a couple reasons:

1. Extensions are intended for extending Data Prepper. Thus, extensions may need to run before parsing a pipeline configuration.
2. The available options in `pipeline_configurations` should be somewhat different from the `extensions`. Not everything in extensions should be configurable in the pipeline.

Reply:
Will rename this part